### ISSUE TITLE
Stripe reconnection results in "There was an error" on recently AT'd site

### ISSUE THREAD

**Issue by cuemarie** [Automattic/jetpack#32637]
### Impacted plugin

Jetpack

### Quick summary

While testing https://github.com/Automattic/jetpack/issues/32614 , I tried disconnecting my site from Stripe and reconnecting it, but I can no longer reconnect.  I tried:

1. Disconnecting/reconnecting
2. Disconnecting and connecting to a new account
3. Deleting all my existing stripe accounts (all pretty old), and starting fresh, but under the same email as before

All these paths result in the error: `There was an error` at the following URL:

URL: `https://public-api.wordpress.com/rest/v1.1/me/stripe_connect/oauth/redirect?scope=read_write&code=ac_OV3QTPxtgTK6AxH9gAJzNMcHor2kxvgV&state=eyJibG9nX2lkIjoxOTQ4MTI0MDQsInVzZXJfaWQiOjIwMjAxODc2NCwibm9uY2UiOiIzOTc3OGM5ZWFjIiwic3JjIjoiY2FseXBzbyJ9`

After my last attempt (number 3 in the list above), I just returned to my WordPress.com dashboard, and I see there that my site is now showing a connection to Stripe under `https://wordpress.com/earn/payments/SITE_ADDRESS`, but the issue I was trying to dis/reconnect to solve (https://github.com/Automattic/jetpack/issues/32614) is still not resolved, so I can't tell if this was successful or not.

### Steps to reproduce

These are the steps I took today:

1. Create a simple site and connect to stripe
2. Set up a page with all the payment blocks
3. AT the site
4. Refresh the page to see that the donations and paid content blocks are missing
5. Disconnect stripe and try to reconnect

### A clear and concise description of what you expected to happen.

Dis/reconnecting should work and return me to my WordPress.com Earn page

### What actually happened

`There was an error`.

![Markup on 2023-08-22 at 15:44:42](https://github.com/Automattic/jetpack/assets/27249804/5e88f63d-ffe5-482f-95a6-7dd8a9db07ef)

### Impact

Some (< 50%)

### Available workarounds?

No and the platform is unusable

### Platform (Simple and/or Atomic)

Atomic

### Logs or notes

It's unclear what exactly is causing this issue, since the error is so vague, so it's possible this is hard to trigger, but this user also continually ran into this error, and I'm not sure they ever got around it:

6699482-zen

User's screenshot:
![Markup on 2023-08-22 at 16:39:51](https://github.com/Automattic/jetpack/assets/27249804/87fe8028-ac2f-42f0-980d-1dab72a09c6a)

**Comment by cuemarie** [comment]
Unfortunately I can't triage this again as it's my account that's the issue ☺️

@Automattic/gold is this something y'all can help look into? Feel free to reprioritize if `high` is not an accurate prioritization. I'm not sure if this happens very often, but it feels difficult to get around once it does happen 🤔

**Comment by millerf** [comment]
I tested it on https://testjetpackissue32637.wordpress.com/5

- Site setup
We can see the button setup on WPCOM simple. Everything seems to work fine.

<img width="300" alt="Screenshot 2023-08-23 at 10 32 36" src="https://github.com/Automattic/jetpack/assets/790558/f2d8274c-4217-41d7-a27b-009a80e878f4">

- Upgraded to eCommerce plan

<img width="300" alt="Screenshot 2023-08-23 at 10 38 37" src="https://github.com/Automattic/jetpack/assets/790558/edfab0a6-5271-4f61-9c0d-121c311a9077">

- Page is now

<img width="300" alt="Screenshot 2023-08-23 at 10 41 08" src="https://github.com/Automattic/jetpack/assets/790558/545d66cb-c3c2-4f29-bd5c-3d5bb17566b9">

- Clicking "log in" let's access the page. Button is still there
<img width="300" alt="Screenshot 2023-08-23 at 10 41 50" src="https://github.com/Automattic/jetpack/assets/790558/4f5c428f-90a1-4e34-a57a-a6e3c1885687">

- After putting the site in "public", the page can be accessed here: https://testjetpackissue32637.wpcomstaging.com/2023/08/23/5/

_Note that the payment button is still there_

<img width="300" alt="Screenshot 2023-08-23 at 10 44 44" src="https://github.com/Automattic/jetpack/assets/790558/447e69f6-fd90-4108-8b5e-d67a033c7ab3">

- Disconnect and reconnect.

**I got the error**


https://github.com/Automattic/jetpack/assets/790558/fd89a258-8f0f-4d32-9551-5a3c9b5b58c8

**Comment by millerf** [comment]
This is a more comprehensive error:

```
[23-Aug-2023 08:56:11 UTC] PHP Fatal error:  Uncaught ArgumentCountError: Too few arguments to function A8C\Billingdaddy\Api\Connected_Accounts\get_stripe_auth_client_details(), 2 passed in /home/wpcom/public_html/public.api/rest/wpcom-json-endpoints/class.wpcom-json-api-memberships.php on line 467 and exactly 3 expected in /home/wpcom/public_html/wp-content/lib/billingdaddy/src/api/connected-accounts-namespace.php:615
```

But I don’t get it. On production the data seems totally fine:
- On the callback (3 params) - class.wpcom-json-api-memberships.php on line 467 
```	
	$data = Connected_Accounts\get_stripe_auth_client_details( $params['code'], $state['user_id'], $state['blog_id'] );
```

And this is totally fine:
```
function get_stripe_auth_client_details( string $code, int $user_id, int $blog_id ) : array {
if ( ! is_configuration_enabled() ) {
return [];
}
return Container::get_stripe_connect_service()->get_auth_client_details( $code, $user_id, $blog_id );
}
```

I am not sure what is going on here...

**Comment by millerf** [comment]
Previous issue seems to be a fluke with Xebug.

I finally got tot he bottom of it.

We are trying to sync a blog option and although it has been merged to `trunk`, it is not yet deployed to Atomic and self-hosted. We'll fix it.

**Comment by millerf** [comment]
@cuemarie I created a temporary fix D119608-code that will allow this to work until Jetpack 12.5 is released.

For the record, the issue was that no self-hosted or Atomic sites were able to finish the Stripe onboarding. A big thanks to the user that reported it!

**Comment by cuemarie** [comment]
Awesome, thank you @millerf ! I'll get that word back to the user!

**Comment by cuemarie** [comment]
Confirming that I retested the steps today, and did not run into the `There was an error` screen. 🎉 

📌 ACTIONS
- Closing as resolved thanks to the temporary fix D119608-code
- Followed up with the impacted user on 6702507-zen
